c2cgeoform/routes.py

```diff
@@ -13,19 +11,25 @@ def table_pregenerator(request, elements, kw):
     def rec_with_pregenerator(route, pattern):
         config.add_route(route, pattern, pregenerator=table_pregenerator)
 
-    global tables
+    tables = []
     for key, mapper in models:
         ca_config = getattr(mapper, '__colanderalchemy_config__', {})
         table = {}
+        # key stand for url path, sometimes designated as table (in match dict)
         table['key'] = key
-        table['name'] = mapper.__tablename__
-        table['title'] = ca_config.get('title', table['name'])
-        table['plural'] = ca_config.get('plural', table['title'])
+        table['title'] = ca_config.get('title', mapper.__tablename__)
+        table['plural'] = ca_config.get('plural', mapper.__tablename__)
         tables.append(table)
 
-    table_regex = '[{}]'.format('|'.join([table['key'] for table in tables]))
+    table_regex = '|'.join(['({})'.format(table['key']) for table in tables])
     base_route = '/{{table:{}}}'.format(table_regex)
 
     rec_with_pregenerator('c2cgeoform_index', '{}/'.format(base_route))
     rec_with_pregenerator('c2cgeoform_grid', '{}/grid.json'.format(base_route))
-    rec_with_pregenerator('c2cgeoform_action', '{}/{{id}}/{{action}}'.format(base_route))
+    rec_with_pregenerator('c2cgeoform_action', '{}/{{id}}/{{action}}'
+                          .format(base_route))
+
+    def add_global(event):
+        event['tables'] = tables
+
+    config.add_subscriber(add_global, iface=BeforeRender)
```

Comment:
If register_models is called many times, you will register the subscriber many times.

Reply:
You cant anyway call register_models manytimes, as it is not possible to register the same route with differents pattern many times.

Reply:
Are you sure about this ? What append if you do it ?

Reply:
I suppose we should add some tests if we want to keep this as a functionnality.

Reply:
pyramid configuration conflict exception or something tasting so

Reply:
Ok so everything seems good on this point